File: Python/AoC2025/days/day1.py
class Day1:
    def __init__(self):
        with open("../../Data/2025/day1.txt") as f:
            self.data = f.readlines()

    def parse_data(self):
        rotations = []
        for line in self.data:
            if line[0] == 'R':
                rotations.append(int(line[1:]))
            elif line[0] == 'L':
                rotations.append(0 - int(line[1:]))
        return rotations

    def part_one(self):
        count = 0
        dial = 50
        rotations = self.parse_data()
        for rotation in rotations:
            dial = (dial + rotation + 100) % 100
            if dial == 0:
                count += 1
        return count
    
    def part_two(self):
        count = 0
        dial = 50
        rotations = self.parse_data()
        for rotation in rotations:
            step = 1 if rotation > 0 else -1
            for _ in range(abs(rotation)):
                dial += step
                if dial == -1:
                    dial = 99
                elif dial == 100:
                    dial = 0
                if dial == 0:
                    count += 1
        return count
    
    def print_results(self):
        print("Day 1:")
        print(f"Part 1: {self.part_one()}")
        print(f"Part 2: {self.part_two()}")

File: JavaScript/AoC2025/days/day1.js
import fs from "fs"
import path from "path"

export default class Day1 {
    constructor() {
        const filePath = path.resolve("../../Data/2025/day1.txt");
        const data = fs.readFileSync(filePath, "utf-8");
        this.lines = data.trim().split("\n");
    }

    printResults() {
        console.log("Day 1:");
        console.log(`Part 1: ${this.part1()}`);
        console.log(`Part 2: ${this.part2()}`);
    }

    part1() {
        let dial = 50;
        let count = 0;
        const rotations = this.lines.map(this.parseRotation.bind(this));
        for (const rotation of rotations) {
            dial = (dial + rotation) % 100;
            if (dial === 0)
                count++;
        }
        return count;
    }

    part2() {
        let dial = 50;
        let count = 0;
        for (const line of this.lines) {
            const step = line[0] === 'R' ? 1 : -1;
            const distance = parseInt(line.slice(1));
            for (let i = 0; i < distance; i++) {
                dial = (100 + dial + step) % 100;
                if (dial === 0)
                    count++;
            }
        }

        return count;
    }

    parseRotation(rotation) {
        const distance = rotation[0] == 'R' 
            ? parseInt(rotation.slice(1)) 
            : - parseInt(rotation.slice(1));
        return distance;
    }
}